pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -568,13 +732,22 @@ func (e *etcd) Destroy(ctx context.Context) error {
 		return err
 	}
 
-	return kutil.DeleteObjects(
+	if err := kutil.DeleteObjects(
 		ctx,
 		e.client,
 		e.emptyHVPA(),
 		e.etcd,
-		e.emptyNetworkPolicy(),
-	)
+		e.emptyNetworkPolicy(ClientNetworkPolicyName),
+		e.emptyNetworkPolicy(PeerNetworkPolicyName),
+	); err != nil {
+		return err
+	}
+
+	if _, ok := e.annotations[v1beta1constants.ShootAlphaControlPlaneHighAvailability]; ok && gardenletfeatures.FeatureGate.Enabled(features.HAControlPlanes) {
+		return kutil.DeleteObject(ctx, e.client, e.emptyNetworkPolicy(PeerNetworkPolicyName))
+	}
```

Comment:
You can probably simply add this `networkPolicy` to the list above, it respects `NotFound` errors